src/tike/ptycho/exitwave.py

```diff
@@ -0,0 +1,134 @@
+"""Functions related to creating and manipulating ptychographic exitwave arrays.
+
+Ptychographic exitwaves are stored as a single complex array which represent
+the wavefield after any and all interaction with the sample and thus there's
+just free space propagation to the detector.
+
+"""
+
+from __future__ import annotations
+import dataclasses
+import logging
+import typing
+
+import cupy as cp
+import numpy as np
+import numpy.typing as npt
+
+import tike.random
+
+logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class ExitWaveOptions:
+    """Manage data and setting related to exitwave updates."""
+
+    noise_model: str = "gaussian"  # { "gaussian", "poisson" }
+
+    step_length_weight: float = 0.5  # 0 <= step_length_weight <= 1
+    step_length_usemodes: str = "all_modes"  # { "dominant_mode", "all_modes" }
+    step_length_start: float = 0.5
+
+    unmeasured_pixels_scaling: float = 0.95
+    unmeasured_pixels: typing.Union[npt.NDArray, None] = dataclasses.field(
+        default_factory=lambda: np.zeros(1, dtype=float))
+    measured_pixels: typing.Union[npt.NDArray, None] = dataclasses.field(
+        default_factory=lambda: np.ones(1, dtype=float))
```

Comment:
Please add docstrings explaining all of these parameters. i.e. These code comments should be converted to docstrings so that they can be loaded into tooltips, the docspage, etc.